### PR TITLE
Team resource

### PR DESCRIPTION
**Pull request by whereandy**
Creates a team with the following TF:
```
resource "vantage_team" "demo_team" {
  name = "Demo Team"
  description = "Demo Team Description"
  user_emails = ["support@vantage.sh"]
}
```

This endpoint accepts either a list of user_emails or a list of user_tokens, but will return 400 if both are supplied.

I'd like to add unit tests, but I'm still getting up to speed on how to write them for this.

### CODE REVIEW

**Review by jmtame — APPROVED** [review]
Nothing jumped out so will give a ✅ but defer to someone else for a more rigorous review if needed

File: vantage/team_resource.go [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+				ElementType:         types.StringType,
+				MarkdownDescription: "User emails.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+				},
+			},
+			"token": schema.StringAttribute{
+				Computed:            true,
```
Comment (conner) [inline vantage/team_resource.go:76]: nit: inconsistent periods in markdown descriptions : )

**Comment by conner** [comment]
FYI I'm looking this over but had to pause to learn what teams were on the platform 😄

**Review by conner — APPROVED** [review]

**Review by conner — APPROVED** [review]

File: examples/resources/vantage_team/provider.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+terraform {
```
Comment (macb) [inline examples/resources/vantage_team/provider.tf:1]: this isn't necessary for the example, just the resource.tf

Comment (macb) [reply examples/resources/vantage_team/provider.tf:1]: Can you run `go generate ./...` on your branch as well. Should update some `docs/` with the new resource.

**Pull request merged**